app/v2/notifications/post_notifications.py

```diff
@@ -30,8 +32,14 @@ def post_notification(notification_type):
     else:
         form = validate(request.get_json(), post_sms_request)
 
+    if service_has_permission(notification_type, authenticated_service.permissions) is False:
+        raise BadRequestError(message="Cannot send {}".format(
+            get_public_notify_type_text(notification_type, plural=True)))
+
     scheduled_for = form.get("scheduled_for", None)
-    service_can_schedule_notification(authenticated_service, scheduled_for)
+    if scheduled_for:
```

Comment:
ok, II've abstracted it out but renamed the method in line with other methods in the validators module